src/Filters/stream_filt.jl

```diff
@@ -627,31 +623,38 @@ function filt!(
     return bufIdx
 end
 
-function filt(self::FIRFilter{Tk}, x::AbstractVector{Tx}) where {Th,Tx,Tk<:FIRKernel{Th}}
-    bufLen         = outputlength(self, length(x))
+function filt(self::FIRFilter, x::AbstractVector)
+    buffer, bufLen = allocate_output(self, x)
+    samplesWritten = filt!(buffer, self, x)
+    return checked_ff_output(buffer, self, bufLen, samplesWritten)
+end
+
+function allocate_output(sf::FIRFilter{Tk}, x::AbstractVector{Tx}) where {Th,Tx,Tk<:FIRKernel{Th}}
     # In some cases when `filt(::FIRFilter{FIRArbitrary}, x)` is called
     # with certain values of `x`, `filt!(buffer, ::FIRFilter{FIRArbitrary}, x)`
     # tries to write one sample too many to the buffer and a `BoundsError`
-    # is thrown.  Add one extra sample to catch these exceptional cases.
+    # is thrown. Add one extra sample to catch these exceptional cases.
     #
     # See https://github.com/JuliaDSP/DSP.jl/issues/317
     #
     # FIXME: Remove this if and when the code in
     #        `filt!(buffer, ::FIRFilter{FIRArbitrary}, x)`
     #        is updated to properly account for pathological arbitrary rates.
+    outLen = outputlength(sf, length(x))
     if Tk <: FIRArbitrary
-        bufLen += 1
+        outLen += 1
     end
-    buffer         = Vector{promote_type(Th,Tx)}(undef, bufLen)
-    samplesWritten = filt!(buffer, self, x)
+    out = Vector{promote_type(Th, Tx)}(undef, outLen)
+    return out, outLen
```

Comment:
No reason to pass `outLen` (aka `bufLen`) around separately instead of just using `length(out)`, I think. But I'm not a huge fan of tacking unrelated changes onto PRs anyway.